sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -183,9 +229,12 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
       // only expand unfoldable children
       val regularAggExprs = aggExpressions
         .filter(e => !e.isDistinct && e.children.exists(!_.foldable))
-      val regularAggChildren = regularAggExprs
+      val regularAggFunChildren = regularAggExprs
         .flatMap(_.aggregateFunction.children.filter(!_.foldable))
-        .distinct
+      val regularAggFilterChildren = regularAggExprs
+        .flatMap(ae => ae.filter.map(_.children.filter(!_.foldable)))
```

Comment:
this looks incorrect. e.g. if the filter is `Multiple(Add(a, 1), 2) > 2`, we should output column `a`, not the first level child `Add(a, 1)`.

I think we should do `.flatMap(_.filter.map(_.reference)).flatten`

Reply:
Very good advice